### ISSUE TITLE
SNOW-149449: 'Unexpected token' uncaught Error

### ISSUE THREAD

**Issue by BenBirt** [snowflakedb/snowflake-connector-nodejs#95]
We occasionally see these errors in a top-level unhandled Error catcher:

```
SyntaxError: Unexpected token <
    at Object.convertRawVariant [as convert] (/app/ts/job-service/job-service-bin.runfiles/npm/node_modules/snowflake-sdk/lib/connection/result/column.js:507:39)
    at Column.extractFromRow (/app/ts/job-service/job-service-bin.runfiles/npm/node_modules/snowflake-sdk/lib/connection/result/column.js:730:21)
    at Column.getRowValue (/app/ts/job-service/job-service-bin.runfiles/npm/node_modules/snowflake-sdk/lib/connection/result/column.js:205:27)
    at Object.getColumnValue (/app/ts/job-service/job-service-bin.runfiles/npm/node_modules/snowflake-sdk/lib/connection/result/chunk.js:348:28)
    at externalizeRow (/app/ts/job-service/job-service-bin.runfiles/npm/node_modules/snowflake-sdk/lib/connection/result/row_stream.js:391:59)
```

I'm not sure what query is running that causes this; we run many on behalf of our customers.

**Comment by nickhartdev** [comment]
We just resolved this issue locally by altering `lib/connection/result/column.js` within our node modules. Our SDK version is 1.6.16.

The query being ran was a simple `SELECT * FROM someTable LIMIT 100` - the table itself had a column with a row that contained XML, which this code seemed to be taking issue with.

To resolve the problem, we simply wrapped the `rawColumnValue` in the middle of the concatenation in `toStringFromVariant` within the `convertRawVariant`.

```
function convertRawVariant(rawColumnValue, column, context)
{
  var ret;
  // if the input is a non-empty string, convert it to a json object
  if (Util.string.isNotNullOrEmpty(rawColumnValue))
  {
    try
    {
      // before: 
      // ret = eval("(" + rawColumnValue + ")"); 
      // after
      ret = eval("(" + toStringFromVariant(rawColumnValue) + ")"); 
    }
    catch (parseError)
    {
      // TODO: log the error
      // throw the error
      throw parseError;
    }
  }
  return ret;
}
```

**Comment by nickhartdev** [comment]
After digging into this further, I'm realizing that while this solution gets around the exception being thrown, it causes some other issues with how JSON is also returned (or any other variant data for that matter). The XML just needs to be returned from this function as is, and everything else should be left with the existing logic.

**Comment by sfc-gh-dszmolka** [comment]
hi - thank you for reporting this issue. i see that despite the long time passed, it's still easily reproducible with the latest version too (1.6.17 at the time of writing). we'll take a look.

**Comment by giolekva** [comment]
What is the status on this issue? We are hitting it as well with sdk version 1.6.7.

**Comment by sfc-gh-dszmolka** [comment]
all updates will be posted here as they are available. sadly, currently there's none :(

**Comment by sfc-gh-dszmolka** [comment]
i have a tiny update here: driver team working on this issue. not timelines still, but will keep this thread updated.

**Comment by sfc-gh-dszmolka** [comment]
#453

**Comment by sfc-gh-dszmolka** [comment]
PR #453 merged. you can either wait for the next release of `snowflake-sdk` to have the fix, or use it even right now by installing the latest code from `master` branch with `https://github.com/snowflakedb/snowflake-connector-nodejs.git`

**Comment by sfc-gh-dszmolka** [comment]
fix released with [1.6.21](https://github.com/snowflakedb/snowflake-connector-nodejs/releases/tag/v1.6.21)
